experimental/class-method-parameter-decorators.md

```diff
@@ -0,0 +1,33 @@
+# [Decorators for Class Method and Constructor Parameters][proposal-class-method-parameter-decorators]
+
+## Decorator
+
+```js
+interface Decorator <: Node {
+    type: "Decorator";
+    expression: Expression;
+}
+```
+
+## FunctionExpression
+
+```js
+interface FunctionExpression <: Function, Expression {
+    type: "FunctionExpression";
+    params: [ Pattern | Parameter ];
+}
+```
+
+If `params` contains a `Parameter` node, its parent must be a `MethodDefinition` under a `ClassBody` node.
```

Comment:
I'm not sure we need this restriction. Could we just say that `Parameter` is always valid in a `Function#params` array? Maybe we could say if `decorators` is not empty, then the parent must be a `MethodDefinition`?

Reply:
```suggestion
If `params` contains a `Parameter` node, its parent must be a `MethodDefinition`.
```

The proposal explicitly disallows decorated parameter in general function expression. Previously I overlooked the spec: I thought `MethodDefinition` is also shared in `ObjectExpression`.

The intent of this PR is to use `Parameter` only when `decorators` are not empty, otherwise we will have two representations for non-decorated parameters. Though I lean towards using the `Parameter` for other elements as well.

Reply:
Gotcha. I suppose we can always revisit later.